lms/static/scripts/frontend_apps/components/GroupConfigSelector.js

```diff
@@ -73,13 +80,115 @@ function GroupSelect({ busy, groupSets, onInput, selectedGroupSetId }) {
   );
 }
 
+/**
+ *
+ *  @typedef GroupSelectHeaderProps
+ * @prop {string} htmlFor
+ * @prop {string} productFamily
+ *
+ * @param {GroupSelectHeaderProps} props
+ */
+function GroupSelectHeader({ htmlFor, productFamily }) {
+  if (productFamily === 'desire2learn') {
+    return <label htmlFor={htmlFor}>Group category: </label>;
+  } else {
+    return <label htmlFor={htmlFor}>Group set: </label>;
+  }
+}
+
+/**
+ *
+ *  @typedef GroupSelectLoadingOptionProps
+ * @prop {string} productFamily
+ *
+ * @param {GroupSelectLoadingOptionProps} props
+ */
+
+function GroupSelectLoadingOption({ productFamily }) {
+  if (productFamily === 'desire2learn') {
+    return <option>Fetching group categories…</option>;
+  } else {
+    return <option>Fetching group sets…</option>;
+  }
+}
+
+/**
+ *
+ *  @typedef GroupSelectPromptOptionProps
+ * @prop {string} productFamily
+ * @prop {boolean} groupSelected
+ *
+ * @param {GroupSelectPromptOptionProps} props
+ */
+
+function GroupSelectPromptOption({ productFamily, groupSelected }) {
+  const groupSetNaming =
+    productFamily === 'desire2learn' ? 'group categories' : 'group set';
+
+  return (
+    <option disabled selected={groupSelected}>
+      Select {groupSetNaming}
+    </option>
+  );
+}
+
 /**
  * ErrorModal shown when the fetched list of course group sets is empty
  *
  * @param {object} props
  *   @param {() => void} props.onCancel
+ *   @param {string} props.productFamily
```

Comment:
Oh so I can declare the props type without giving the type a name, learner that too late in the file.